packages/engine/lib/orchestrator/src/manifest.rs

```diff
@@ -0,0 +1,579 @@
+use std::{
+    borrow::Borrow,
+    collections::HashMap,
+    fs::File,
+    io::BufReader,
+    path::{Path, PathBuf},
+};
+
+use error::{bail, ensure, report, Result, ResultExt};
+use hash_engine::{
+    fetch::parse_raw_csv_into_json,
+    proto::{
+        ExperimentRun, ExperimentRunBase, InitialState, InitialStateName, ProjectBase,
+        SharedBehavior, SharedDataset, SimPackageArgs,
+    },
+};
+use serde::{self, de::DeserializeOwned};
+use serde_json::Value as SerdeValue;
+
+use crate::ExperimentType;
+
+const BEHAVIOR_FILE_EXTENSIONS: [&str; 3] = ["js", "py", "rs"];
+const DATASET_FILE_EXTENSIONS: [&str; 2] = ["csv", "json"];
+
+#[derive(Debug, Default, Clone)]
+pub struct Manifest {
+    pub initial_state: Option<InitialState>,
+    pub behaviors: Vec<SharedBehavior>,
+    pub datasets: Vec<SharedDataset>,
+    pub globals_json: Option<String>,
+    pub analysis_json: Option<String>,
+    pub experiments_json: Option<String>,
+    pub dependencies: HashMap<String, SerdeValue>,
+}
+
+impl Manifest {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn set_initial_state_from_file<P: AsRef<Path>>(
+        &mut self,
+        path: P,
+    ) -> Result<Option<InitialState>> {
+        let path = path.as_ref();
+        ensure!(path.is_file(), "Not a behavior file: {path:?}");
+
+        Ok(self.initial_state.replace(InitialState {
+            name: match file_extension(&path)?.as_str() {
+                "js" => InitialStateName::InitJs,
+                "py" => InitialStateName::InitPy,
+                "json" => InitialStateName::InitJson,
+                _ => bail!("Not a valid initial state file: {path:?}"),
+            },
+            src: file_contents(path)?,
+        }))
+    }
+
+    pub fn set_initial_state_from_files<P1: AsRef<Path>, P2: AsRef<Path>, P3: AsRef<Path>>(
+        &mut self,
+        js_path: P1,
+        py_path: P2,
+        json_path: P3,
+    ) -> Result<Option<InitialState>> {
+        let js_path = js_path.as_ref();
+        let py_path = py_path.as_ref();
+        let json_path = json_path.as_ref();
+
+        debug!("Reading initial state files");
+        if js_path.is_file() {
+            if py_path.is_file() {
+                warn!("{py_path:?} was supplied with {js_path:?}, ignoring {py_path:?}");
```

Comment:
I actually think this is less readable than before. 
```"init.py was supplied with init.js, ignoring init.py"```
Open to discussion on that obviously, perhaps giving the exact paths is better in some cases? I just think we were being a lot clearer that js > py > json

Otherwise this is going to be such a long warning message like 
/users/bob/documents/this/is/some/path/to/my/init/folder/foo/bar/baz/init.py was supplied with /users/bob/documents/this/is/some/path/to/my/init/folder/foo/bar/baz/init.js, ignoring /users/bob/documents/this/is/some/path/to/my/init/folder/foo/bar/baz/init.py

Reply:
Yeah, that's true. As this is not only used for `init.py` etc., I'd remove the path information and just warn
```rust
warn!("Multiple initial states were provided, using {js_path:?}");
```

Reply:
Can you explain how it's not just used for `init.py`? As the function is supposed to encode the specific hierarchy we expect (js > py > json)

Reply:
The function is called with `manifest.set_initial_state_from_files("path/to/init.js", "path/to/init.py", "path/to/init.json")?`. For the CLI, those are `init.js`, `init.py`, and `init.json`, but any files could be passed.

Reply:
We could move out this function to the CLI again, instead, and use the old warning.

Reply:
I changed the code to use `init.js`, `init.py`, and `init.json`. This function is currently only used by `from_local`, it's still possible to specify a init file in another way.